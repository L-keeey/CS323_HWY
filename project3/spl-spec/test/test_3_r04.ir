FUNCTION sqr :
PARAM v1
t2 := v1
t3 := v1
t1 := t2 * t3
RETURN t1
FUNCTION cbc :
PARAM v2
t7 := v2
t8 := v2
t5 := t7 * t8
t6 := v2
t4 := t5 * t6
RETURN t4
FUNCTION main :
READ t10
&v3 := t10
t9 := &v3
t11 := &v3
WRITE t11
ARG &v3
t13 := CALL sqr
&v4 := t13
t12 := &v4
t14 := &v4
WRITE t14
ARG &v3
t16 := CALL cbc
&v4 := t16
t15 := &v4
t17 := &v4
WRITE t17
t18 := #0
RETURN t18
