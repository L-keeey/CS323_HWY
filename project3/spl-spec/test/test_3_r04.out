FunDec
VarDec
RETURN STMT
FunDec
VarDec
RETURN STMT
FunDec
Dec newvar
